diff --git a/vendor/libpng/pngrutil.c b/vendor/libpng/pngrutil.c
index a4eda54c8c..a008ee685f 100644
--- a/vendor/libpng/pngrutil.c
+++ b/vendor/libpng/pngrutil.c
@@ -446,7 +446,7 @@ png_inflate_claim(png_structrp png_ptr, png_uint_32 owner)
     */
    {
       int ret; /* zlib return code */
-#if ZLIB_VERNUM >= 0x1240
+#if 0
       int window_bits = 0;
 
 # if defined(PNG_SET_OPTION_SUPPORTED) && defined(PNG_MAXIMUM_INFLATE_WINDOW)
@@ -475,7 +475,7 @@ png_inflate_claim(png_structrp png_ptr, png_uint_32 owner)
 
       if ((png_ptr->flags & PNG_FLAG_ZSTREAM_INITIALIZED) != 0)
       {
-#if ZLIB_VERNUM >= 0x1240
+#if 0
          ret = inflateReset2(&png_ptr->zstream, window_bits);
 #else
          ret = inflateReset(&png_ptr->zstream);
@@ -484,7 +484,7 @@ png_inflate_claim(png_structrp png_ptr, png_uint_32 owner)
 
       else
       {
-#if ZLIB_VERNUM >= 0x1240
+#if 0
          ret = inflateInit2(&png_ptr->zstream, window_bits);
 #else
          ret = inflateInit(&png_ptr->zstream);
