diff --git a/vendor/libpng/pngpriv.h b/vendor/libpng/pngpriv.h
index f91666c700..59275a2105 100644
--- a/vendor/libpng/pngpriv.h
+++ b/vendor/libpng/pngpriv.h
@@ -1023,8 +1023,8 @@
 /* Validate the include paths - the include path used to generate pnglibconf.h
  * must match that used in the build, or we must be using pnglibconf.h.prebuilt:
  */
-#if PNG_ZLIB_VERNUM != 0 && PNG_ZLIB_VERNUM != ZLIB_VERNUM
-#  error The include path of <zlib.h> is incorrect
+// #if PNG_ZLIB_VERNUM != 0 && PNG_ZLIB_VERNUM != ZLIB_VERNUM
+// #  error The include path of <zlib.h> is incorrect
    /* When pnglibconf.h was built, the copy of zlib.h that it used was not the
     * same as the one being used here.  Considering how libpng makes decisions
     * to use the zlib API based on the zlib version number, the -I options must
@@ -1034,7 +1034,7 @@
     * CFLAGS, which is unlikely to work.  All the preprocessor options, and all
     * the -I options in particular, should be in CPPFLAGS.
     */
-#endif
+// #endif
 
 /* This is used for 16-bit gamma tables -- only the top level pointers are
  * const; this could be changed:
diff --git a/vendor/libpng/pngstruct.h b/vendor/libpng/pngstruct.h
index fe5fa04155..b3e98ec5a7 100644
--- a/vendor/libpng/pngstruct.h
+++ b/vendor/libpng/pngstruct.h
@@ -24,7 +24,7 @@
    /* We must ensure that zlib uses 'const' in declarations. */
 #  define ZLIB_CONST
 #endif
-#include "zlib.h"
+#include "../miniz/miniz.h"
 #ifdef const
    /* zlib.h sometimes #defines const to nothing, undo this. */
 #  undef const
