diff --git a/vendor/pdfio/pdfio-content.c b/vendor/pdfio/pdfio-content.c
index f71277d444..840f251d12 100644
--- a/vendor/pdfio/pdfio-content.c
+++ b/vendor/pdfio/pdfio-content.c
@@ -3270,7 +3270,7 @@ copy_png(pdfio_dict_t *dict,		// I - Dictionary
       return (NULL);
     }
 
-    temp = (unsigned)((buffer[0] << 24) | (buffer[1] << 16) | (buffer[2] << 8) | buffer[3]);
+    temp = ((unsigned)buffer[0] << 24) | ((unsigned)buffer[1] << 16) | ((unsigned)buffer[2] << 8) | (unsigned)buffer[3];
     if (temp != crc)
     {
       pdfioStreamClose(st);
