diff --git a/vendor/pdfio/pdfio-dict.c b/vendor/pdfio/pdfio-dict.c
index 95a1a365a1..aa4c08f9ce 100644
--- a/vendor/pdfio/pdfio-dict.c
+++ b/vendor/pdfio/pdfio-dict.c
@@ -14,7 +14,7 @@
 // Local functions...
 //
 
-static int	compare_pairs(_pdfio_pair_t *a, _pdfio_pair_t *b);
+static int	compare_pairs(const void *a, const void *b);
 
 
 //
@@ -42,7 +42,7 @@ pdfioDictClear(pdfio_dict_t *dict,	// I - Dictionary
   {
     pkey.key = key;
 
-    if ((pair = (_pdfio_pair_t *)bsearch(&pkey, dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), (int (*)(const void *, const void *))compare_pairs)) != NULL)
+    if ((pair = (_pdfio_pair_t *)bsearch(&pkey, dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), compare_pairs)) != NULL)
     {
       // Yes, remove it...
       if (pair->value.type == PDFIO_VALTYPE_BINARY)
@@ -563,7 +563,7 @@ _pdfioDictGetValue(pdfio_dict_t *dict,	// I - Dictionary
 
   temp.key = key;
 
-  if ((match = bsearch(&temp, dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), (int (*)(const void *, const void *))compare_pairs)) != NULL)
+  if ((match = bsearch(&temp, dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), compare_pairs)) != NULL)
   {
     PDFIO_DEBUG("_pdfioDictGetValue: Match, returning ");
     PDFIO_DEBUG_VALUE(&(match->value));
@@ -1020,7 +1020,7 @@ _pdfioDictSetValue(
 
     pkey.key = key;
 
-    if ((pair = (_pdfio_pair_t *)bsearch(&pkey, dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), (int (*)(const void *, const void *))compare_pairs)) != NULL)
+    if ((pair = (_pdfio_pair_t *)bsearch(&pkey, dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), compare_pairs)) != NULL)
     {
       // Yes, replace the value...
       PDFIO_DEBUG("_pdfioDictSetValue: Replacing existing value.\n");
@@ -1055,7 +1055,7 @@ _pdfioDictSetValue(
 
   // Re-sort the dictionary and return...
   if (dict->num_pairs > 1 && compare_pairs(pair - 1, pair) > 0)
-    qsort(dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), (int (*)(const void *, const void *))compare_pairs);
+    qsort(dict->pairs, dict->num_pairs, sizeof(_pdfio_pair_t), compare_pairs);
 
 #ifdef DEBUG
   PDFIO_DEBUG("_pdfioDictSetValue(%p): %lu pairs\n", (void *)dict, (unsigned long)dict->num_pairs);
@@ -1116,8 +1116,10 @@ _pdfioDictWrite(pdfio_dict_t *dict,	// I - Dictionary
 //
 
 static int				// O - Result of comparison
-compare_pairs(_pdfio_pair_t *a,		// I - First pair
-              _pdfio_pair_t *b)		// I - Second pair
+compare_pairs(const void *a,		// I - First pair
+              const void *b)		// I - Second pair
 {
-  return (strcmp(a->key, b->key));
+  const _pdfio_pair_t *p = (const _pdfio_pair_t *)a;
+  const _pdfio_pair_t *q = (const _pdfio_pair_t *)b;
+  return (strcmp(p->key, q->key));
 }
